Mathlib/Algebra/Polynomial/RingDivision.lean

```diff
@@ -455,35 +104,10 @@ theorem rootMultiplicity_eq_rootMultiplicity {p : R[X]} {t : R} :
   convert (multiplicity_map_eq <| algEquivAevalXAddC t).symm using 2
   simp [C_eq_algebraMap]
 
-theorem rootMultiplicity_eq_natTrailingDegree' {p : R[X]} :
-    p.rootMultiplicity 0 = p.natTrailingDegree := by
-  by_cases h : p = 0
-  · simp only [h, rootMultiplicity_zero, natTrailingDegree_zero]
-  refine le_antisymm ?_ ?_
-  · rw [rootMultiplicity_le_iff h, map_zero, sub_zero, X_pow_dvd_iff, not_forall]
-    exact ⟨p.natTrailingDegree,
-      fun h' ↦ trailingCoeff_nonzero_iff_nonzero.2 h <| h' <| Nat.lt.base _⟩
-  · rw [le_rootMultiplicity_iff h, map_zero, sub_zero, X_pow_dvd_iff]
-    exact fun _ ↦ coeff_eq_zero_of_lt_natTrailingDegree
-
 theorem rootMultiplicity_eq_natTrailingDegree {p : R[X]} {t : R} :
```

Comment:
This one and the previous clearly have strong commonalities, if we can't keep them together then let's be sure to add docstrings referencing the other